### PR TITLE
Add viewport directive to HTML templates to optimise for mobile

### PR DESCRIPTION
**Pull request by gsouquet**
This adds a meta viewport tag to the HTML templates so that the documents are scaled properly when viewed on mobile. And because an image is worth a thousand words, here's a GIF that illustrates the current problem

validate-email

Signed-off-by: Germain Souquet <germains@element.io>

### Pull Request Checklist

* [X] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

File: synapse/res/templates/add_threepid_failure.html [hunk 1]
```diff
@@ -1,8 +1,13 @@
-<html>
-<head></head>
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>request failed</title>
```
Comment (DMRobertson) [inline synapse/res/templates/add_threepid_failure.html:7]:
```suggestion
    <title>Request failed</title>
```

File: synapse/res/templates/password_reset_confirmation.html [hunk 2]
```diff
@@ -1,5 +1,8 @@
-<html>
-<head></head>
+<html lang="en">
+<head>
```
Comment (DMRobertson) [inline synapse/res/templates/password_reset_confirmation.html:2]: Should this have a title? (Not sure if this is intended for use in emails)

The same for `password_reset_failure` and `password_reset_success` too.

File: synapse/res/templates/registration_failure.html [hunk 3]
```diff
@@ -1,5 +1,8 @@
-<html>
-<head></head>
+<html lang="en">
+<head>
```
Comment (DMRobertson) [inline synapse/res/templates/registration_failure.html:2]: Should this have a `<title>`?

**Review by DMRobertson — CHANGES_REQUESTED** [review]

**Comment by DMRobertson** [comment]
Otherwise LGTM!

File: changelog.d/13493.misc [hunk 4]
```diff
@@ -0,0 +1 @@
+Add viewport directive to HTML templates.
```
Comment (anoadragon453) [inline changelog.d/13493.misc:1]: We try to keep our changelogs fairly sysadmin-friendly.

```suggestion
Modify HTML template content to better support mobile devices screen sizes.
```

**Review by anoadragon453 — CHANGES_REQUESTED** [review]
I've noticed while looking through that some pages set `user-scalable=no` in their viewport directive. This prevents users from zooming in on a page, and is considered bad for accessibility. We should fix this as well, though happy for it to be in a separate PR.

File: changelog.d/13493.misc [hunk 5]
```diff
@@ -1 +1 @@
-Add viewport directive to HTML templates.
\ No newline at end of file
+Modify HTML template content to better support mobile devices screen sizes.
```
Comment (DMRobertson) [inline changelog.d/13493.misc:1]:
```suggestion
Modify HTML template content to better support mobile devices' screen sizes.
```

**Review by DMRobertson — APPROVED** [review]
Otherwise LGTM!

**Review by anoadragon453 — APPROVED** [review]

**Pull request merged**